Dia 1 
Objetivo del analisis: Analizar los datos para ver la posibilidad de dónde abrir una escuela de buceo en los 3 lugares menos peligrosos. 

In [78]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.0.1 -> 26.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [79]:
#1. Download dataset and import it into Python.

import pandas as pd

#url = "https://raw.githubusercontent.com/roizherrerapilar-hub/Mini-project---Shark-Attacks/main/Shark_Attacks_Dataset.csv"

df = pd.read_csv('Shark_Attacks_Dataset.csv', sep=';')
pd.set_option('display.max_columns', None) #Con esta funcion vemos todas las columnas que tiene el archivo



C:\Users\MEL\AppData\Local\Temp\ipykernel_5260\2278032266.py:7: DtypeWarning: Columns (0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('Shark_Attacks_Dataset.csv', sep=';')


In [80]:
#2. Visualize the Dataframe 

df.head() 

,Date,Year,Type,Country,State,Location,Activity,Name,Sex,Age,Injury,Fatal Y/N,Time,Species,Source,pdf,href formula,href,Case Number,Case Number.1,original order,Unnamed: 21,Unnamed: 22,Unnamed: 23,Unnamed: 24,Unnamed: 25,Unnamed: 26,Unnamed: 27,Unnamed: 28,Unnamed: 29,Unnamed: 30,Unnamed: 31,Unnamed: 32,Unnamed: 33,Unnamed: 34,Unnamed: 35,Unnamed: 36,Unnamed: 37,Unnamed: 38,Unnamed: 39,Unnamed: 40,Unnamed: 41,Unnamed: 42,Unnamed: 43,Unnamed: 44,Unnamed: 45,Unnamed: 46,Unnamed: 47,Unnamed: 48,Unnamed: 49,Unnamed: 50,Unnamed: 51,Unnamed: 52,Unnamed: 53,Unnamed: 54,Unnamed: 55,Unnamed: 56,Unnamed: 57,Unnamed: 58,Unnamed: 59,Unnamed: 60,Unnamed: 61,Unnamed: 62,Unnamed: 63,Unnamed: 64,Unnamed: 65,Unnamed: 66,Unnamed: 67,Unnamed: 68,Unnamed: 69,Unnamed: 70,Unnamed: 71,Unnamed: 72,Unnamed: 73,Unnamed: 74,Unnamed: 75,Unnamed: 76,Unnamed: 77,Unnamed: 78,Unnamed: 79,Unnamed: 80,Unnamed: 81,Unnamed: 82,Unnamed: 83,Unnamed: 84,Unnamed: 85,Unnamed: 86,Unnamed: 87,Unnamed: 88,Unnamed: 89,Unnamed: 90,Unnamed: 91,Unnamed: 92,Unnamed: 93,Unnamed: 94,Unnamed: 95,Unnamed: 96,Unnamed: 97,Unnamed: 98,Unnamed: 99,Unnamed: 100,Unnamed: 101,Unnamed: 102,Unnamed: 103,Unnamed: 104,Unnamed: 105,Unnamed: 106,Unnamed: 107,Unnamed: 108,Unnamed: 109,Unnamed: 110,Unnamed: 111,Unnamed: 112,Unnamed: 113,Unnamed: 114,Unnamed: 115,Unnamed: 116,Unnamed: 117,Unnamed: 118,Unnamed: 119,Unnamed: 120,Unnamed: 121,Unnamed: 122,Unnamed: 123,Unnamed: 124,Unnamed: 125,Unnamed: 126,Unnamed: 127,Unnamed: 128,Unnamed: 129,Unnamed: 130,Unnamed: 131,Unnamed: 132,Unnamed: 133,Unnamed: 134,Unnamed: 135,Unnamed: 136,Unnamed: 137,Unnamed: 138,Unnamed: 139,Unnamed: 140,Unnamed: 141,Unnamed: 142,Unnamed: 143,Unnamed: 144,Unnamed: 145,Unnamed: 146,Unnamed: 147,Unnamed: 148,Unnamed: 149,Unnamed: 150,Unnamed: 151,Unnamed: 152,Unnamed: 153,Unnamed: 154,Unnamed: 155,Unnamed: 156,Unnamed: 157,Unnamed: 158,Unnamed: 159,Unnamed: 160,Unnamed: 161,Unnamed: 162,Unnamed: 163,Unnamed: 164,Unnamed: 165,Unnamed: 166,Unnamed: 167,Unnamed: 168,Unnamed: 169,Unnamed: 170,Unnamed: 171,Unnamed: 172,Unnamed: 173,Unnamed: 174,Unnamed: 175,Unnamed: 176,Unnamed: 177,Unnamed: 178,Unnamed: 179,Unnamed: 180,Unnamed: 181,Unnamed: 182,Unnamed: 183,Unnamed: 184,Unnamed: 185,Unnamed: 186,Unnamed: 187,Unnamed: 188,Unnamed: 189,Unnamed: 190,Unnamed: 191,Unnamed: 192,Unnamed: 193,Unnamed: 194,Unnamed: 195,Unnamed: 196,Unnamed: 197,Unnamed: 198,Unnamed: 199,Unnamed: 200,Unnamed: 201,Unnamed: 202,Unnamed: 203,Unnamed: 204,Unnamed: 205,Unnamed: 206,Unnamed: 207,Unnamed: 208,Unnamed: 209,Unnamed: 210,Unnamed: 211,Unnamed: 212,Unnamed: 213,Unnamed: 214,Unnamed: 215,Unnamed: 216,Unnamed: 217,Unnamed: 218,Unnamed: 219,Unnamed: 220,Unnamed: 221,Unnamed: 222,Unnamed: 223,Unnamed: 224,Unnamed: 225,Unnamed: 226,Unnamed: 227,Unnamed: 228,Unnamed: 229,Unnamed: 230,Unnamed: 231,Unnamed: 232,Unnamed: 233,Unnamed: 234,Unnamed: 235,Unnamed: 236,Unnamed: 237,Unnamed: 238,Unnamed: 239,Unnamed: 240,Unnamed: 241,Unnamed: 242,Unnamed: 243,Unnamed: 244,Unnamed: 245,Unnamed: 246,Unnamed: 247,Unnamed: 248,Unnamed: 249,Unnamed: 250,Unnamed: 251,Unnamed: 252,Unnamed: 253,Unnamed: 254
0,14/04,2026,UNprovoked,Maldives,Gaafu Alif Atoll,Kooddoo,Swimming,Not stated - on honeymoon,M,?,Leg strpped off flesh later amputated in hospital,N,?,Unknown,The U.S. Sun: Simon De Marchi,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN

In [81]:
#3. Indicate the number of columns and rows. 
df.shape 

(78395, 255)

In [82]:
#4. Delete the columns with null values (from column 16: source)
df = df.drop(columns=df.columns[15:]) 
df.head()

,Date,Year,Type,Country,State,Location,Activity,Name,Sex,Age,Injury,Fatal Y/N,Time,Species,Source
0,14/04,2026,UNprovoked,Maldives,Gaafu Alif Atoll,Kooddoo,Swimming,Not stated - on honeymoon,M,?,Leg strpped off flesh later amputated in hospital,N,?,Unknown,The U.S. Sun: Simon De Marchi
1,3rd April,2026,Unprovoked,Australia,South Australia,Middleton Beach Fleurieu Peninsula Adelaide,Surfing,Oliver Tokic-Bensley,M,16,Bite wound to R ankle,N,?,Bronze Whaler,ABC News: The Guardian: Andrew Currie and Bob...
2,26th March,2026,Unprovoked,Bahamas,Andros Island,Fresh Creek,Swimming,Australian woman,F,22,Severe injuries to R arm,N,1730hrs,Unknown,WIC News: Melissa Michaelson
3,25th March,2026,Unprovoked,Australia,South Australia,Cape Jaffa Limestone Coast,Diving,Luke Kuhn,M,?,NaN,N,?,Great White Shark 3.5m (11.5ft),Bob Myatt GSAF
4,22nd-23rd March,2026,Unprovoked,Australia,NSW,Little Avalon Beach,Surfing,Unknown,M,30+,Graze injuries R leg and ankle,N,?,Unknown,Melissa Michaelson: Instagram


In [83]:
#5. Indicate the names of the columns. 
df.columns 

Index(['Date', 'Year', 'Type', 'Country', 'State', 'Location', 'Activity',
       'Name', 'Sex', 'Age', 'Injury', 'Fatal Y/N', 'Time', 'Species ',
       'Source'],
      dtype='object')

In [84]:
#6. Delete the rows with null values. 
df.dropna()
df.head()

,Date,Year,Type,Country,State,Location,Activity,Name,Sex,Age,Injury,Fatal Y/N,Time,Species,Source
0,14/04,2026,UNprovoked,Maldives,Gaafu Alif Atoll,Kooddoo,Swimming,Not stated - on honeymoon,M,?,Leg strpped off flesh later amputated in hospital,N,?,Unknown,The U.S. Sun: Simon De Marchi
1,3rd April,2026,Unprovoked,Australia,South Australia,Middleton Beach Fleurieu Peninsula Adelaide,Surfing,Oliver Tokic-Bensley,M,16,Bite wound to R ankle,N,?,Bronze Whaler,ABC News: The Guardian: Andrew Currie and Bob...
2,26th March,2026,Unprovoked,Bahamas,Andros Island,Fresh Creek,Swimming,Australian woman,F,22,Severe injuries to R arm,N,1730hrs,Unknown,WIC News: Melissa Michaelson
3,25th March,2026,Unprovoked,Australia,South Australia,Cape Jaffa Limestone Coast,Diving,Luke Kuhn,M,?,NaN,N,?,Great White Shark 3.5m (11.5ft),Bob Myatt GSAF
4,22nd-23rd March,2026,Unprovoked,Australia,NSW,Little Avalon Beach,Surfing,Unknown,M,30+,Graze injuries R leg and ankle,N,?,Unknown,Melissa Michaelson: Instagram


In [85]:
#7. Basic overview of the dataset. 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78395 entries, 0 to 78394
Data columns (total 15 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Date       14177 non-null  object
 1   Year       14171 non-null  object
 2   Type       14139 non-null  object
 3   Country    14075 non-null  object
 4   State      13201 non-null  object
 5   Location   13041 non-null  object
 6   Activity   13009 non-null  object
 7   Name       13739 non-null  object
 8   Sex        13019 non-null  object
 9   Age        8187 non-null   object
 10  Injury     14103 non-null  object
 11  Fatal Y/N  13053 non-null  object
 12  Time       7121 non-null   object
 13  Species    7913 non-null   object
 14  Source     14135 non-null  object
dtypes: object(15)
memory usage: 9.0+ MB


Comentarios:
- Vemos que el data type de fecha es string y deberia ser data type object. 
- Los valores no nulos de las columnas: date, year, type, country, injury, source tienen valores similares. 
- Vemos que las columnas de age y time presentan en algunas filas un signo de interrogacion por falta de datos, por lo que mas adelante analizaremos que ocurre. 


Clasificacion de las columnas: 
- Categorical: type, country, state, location, activity, name, sex, injury, fatal Y/N, species, source. 
- Numerical: date, year, age, time. 

In [86]:
#8. Indicate the names of the columns.
print(df.columns.tolist())  


['Date', 'Year', 'Type', 'Country', 'State', 'Location', 'Activity', 'Name', 'Sex', 'Age', 'Injury', 'Fatal Y/N', 'Time', 'Species ', 'Source']


In [87]:
#9. Analysis of categorical variables. 
df.describe()

,Date,Year,Type,Country,State,Location,Activity,Name,Sex,Age,Injury,Fatal Y/N,Time,Species,Source
count,14177,14171,14139,14075,13201,13041,13009,13739,13019,8187,14103,13053,7121,7913,14135
unique,6099,303,15,254,949,4632,1613,5802,12,179,4198,13,478,1747,5414
top,1957,2015,Unprovoked,USA,Florida,"New Smyrna Beach, Volusia County",Surfing,male,M,17,FATAL,N,Afternoon,White shark,"K. McMurray, TrackingSharks.com"
freq,22,282,10474,5160,2384,382,2302,1358,11364,348,1726,9886,430,388,262


Comentarios: 
- Vemos que la columna tiene presenta solo 14 valores unicos provocados siendo el valor total de 5413. Lo cual podemos deducir que hay algo raro que debemos de analizar posteriormente. 
- Vemos que la columna de country tiene 253 valores unicos, siendo el mas dominante, USA. Por lo tanto, se deduce que los ataques se concentran en USA. 
- Nos resulta sospechoso ver que el valor mas frecuente en la columna de date sea 1957, dado que no tiene formato de fecha mm/yy/dd. 
- En la columna de especies, nos llama la atencion que siendo 1746 especies unicas White Shark es la especie con mas ataques registrados en las costas de Florida. 

# Día 2 - Data Cleaning Process
**Objetivo:** Limpiar el dataset de ataques de tiburón para identificar los 3 lugares menos peligrosos donde abrir una franquicia de buceo.

**Secciones:**
1. Manejo de valores nulos
2. Manejo de duplicados
3. Manipulación de cadenas de texto
4. Uso de regex
5. Formato de datos
6. Verificación final


In [88]:
#1. Valores nulos
#Hemos eliminado en la primera parte todas las columnas y filas con valores nulos. 

In [89]:
# Count the number of null values in each column
df.isna().sum()

Date         64218
Year         64224
Type         64256
Country      64320
State        65194
Location     65354
Activity     65386
Name         64656
Sex          65376
Age          70208
Injury       64292
Fatal Y/N    65342
Time         71274
Species      70482
Source       64260
dtype: int64

In [90]:
#2. Valores duplicados
df = df.drop_duplicates()
df.head()

,Date,Year,Type,Country,State,Location,Activity,Name,Sex,Age,Injury,Fatal Y/N,Time,Species,Source
0,14/04,2026,UNprovoked,Maldives,Gaafu Alif Atoll,Kooddoo,Swimming,Not stated - on honeymoon,M,?,Leg strpped off flesh later amputated in hospital,N,?,Unknown,The U.S. Sun: Simon De Marchi
1,3rd April,2026,Unprovoked,Australia,South Australia,Middleton Beach Fleurieu Peninsula Adelaide,Surfing,Oliver Tokic-Bensley,M,16,Bite wound to R ankle,N,?,Bronze Whaler,ABC News: The Guardian: Andrew Currie and Bob...
2,26th March,2026,Unprovoked,Bahamas,Andros Island,Fresh Creek,Swimming,Australian woman,F,22,Severe injuries to R arm,N,1730hrs,Unknown,WIC News: Melissa Michaelson
3,25th March,2026,Unprovoked,Australia,South Australia,Cape Jaffa Limestone Coast,Diving,Luke Kuhn,M,?,NaN,N,?,Great White Shark 3.5m (11.5ft),Bob Myatt GSAF
4,22nd-23rd March,2026,Unprovoked,Australia,NSW,Little Avalon Beach,Surfing,Unknown,M,30+,Graze injuries R leg and ankle,N,?,Unknown,Melissa Michaelson: Instagram


In [114]:
df.shape 

#Vemos que el dataframe se ha reducido de (39196, 255) a (7087, 15)

(4013, 9)

In [115]:
#3. Manipulacion de strings
#Poner los nombres de las columnas en minuscula
df.columns = df.columns.str.lower()
df.head()

,country,state,location,activity,age,injury,species,age_range,fatal
1,Australia,South Australia,Middleton Beach Fleurieu Peninsula Adelaide,Surfing,16,Bite wound to R ankle,Bronze Whaler,<18,False
2,Bahamas,Andros Island,Fresh Creek,Swimming,22,Severe injuries to R arm,Unknown,18-25,False
4,Australia,NSW,Little Avalon Beach,Surfing,30,Graze injuries R leg and ankle,Unknown,26-35,False
5,USA,California,Big River Beach Mendocino County,Surfing,39,Injuries to both legs and feet,Great White Shark,36-50,False
8,Australia,Queensland,Lady Elliott Island,Snorkeling,50,Lacerations elbow and abdomen,Unknown,36-50,False


In [ ]:
#Columnas que consideramos importantes para nuestro proyecto: type, country, state, location, activity, age, injury, species. 
#4. Eliminacion de las columnas innecesarias:
df = df.drop(columns=['date', 'year', 'type', 'name', 'sex', 'fatal y/n', 'time', 'source'])
df.head()

In [94]:
#5. Clasificar las edades en rango. 
age_values = df['age'].tolist()
print(age_values)

['?', '16', '22', '?', '30+', '39', '?', '?', "50's", '55', '?', '13', '?', '?', '39', '11', '27', '12', '26', '?', '56', '?', '?', '?', '?', '55', '?', '24', '26', '25', '61', '40', '?', '?', '13', '14', '14', '50+', '54', '48', '57', '8', '8', '?', '63', '?', '9', '39', '19', '7', '85', '?', '69', '18', '66', '21', '40', '37', '16', '20', '?', '12', '9', '42', '26', '14', '66', '26', '45', '20', '30', '?', '30+', '37', '56', '40+', '29', '35', '58', '29', '24', '20', '55', '17', '12', '37', '36', '23', '40', '28', '?', '69', '48', '40', '60+', '?', '57', '24', '45', '61', '57', '27', '38', '16', '68', '33', '30', '16', '15', '23', '41', '14', '43', '26', 'Middle age', '18', '18', '21', '?', '49', '25', '15', '45', '46', '23', '19', '27', '20', '65', '64', '24', '16', '26', '?', '13', nan, '11', '46', '32', nan, '29', '10', '64', '62', '39', '22', '15', '16', '46', '21', '66', '52', '20', '44', '26', '21', '47', '32', '66', nan, nan, '66', '55', '30', '59', nan, '8', '50', '52', '45',

In [95]:
try:
    df = pd.read_csv('Shark_Attacks_Dataset.csv', sep=';', low_memory=False)
    print("✅ Cargado: Shark_Attacks_Dataset.csv")
except FileNotFoundError:
    df = pd.read_csv('Dataset_CSV.csv', sep=';', low_memory=False)
    print("✅ Cargado: Dataset_CSV.csv")

✅ Cargado: Shark_Attacks_Dataset.csv


In [96]:
df = df.drop(columns=df.columns[15:])
df.columns = df.columns.str.lower()
print(df.columns.tolist())

['date', 'year', 'type', 'country', 'state', 'location', 'activity', 'name', 'sex', 'age', 'injury', 'fatal y/n', 'time', 'species ', 'source']


In [97]:
df = df.drop(columns=df.columns[15:])
df.columns = df.columns.str.lower().str.strip()

In [98]:
# Eliminar solo las columnas que existan
cols_to_drop = ['date', 'year', 'type', 'name', 'sex', 'fatal y/n', 'time', 'source']
df = df.drop(columns=[c for c in cols_to_drop if c in df.columns])
df = df.drop_duplicates().reset_index(drop=True)

In [99]:
print(f'Shape de partida: {df.shape}')
print(f'Columnas: {df.columns.tolist()}')
df.head()

Shape de partida: (7053, 7)
Columnas: ['country', 'state', 'location', 'activity', 'age', 'injury', 'species']


,country,state,location,activity,age,injury,species
0,Maldives,Gaafu Alif Atoll,Kooddoo,Swimming,?,Leg strpped off flesh later amputated in hospital,Unknown
1,Australia,South Australia,Middleton Beach Fleurieu Peninsula Adelaide,Surfing,16,Bite wound to R ankle,Bronze Whaler
2,Bahamas,Andros Island,Fresh Creek,Swimming,22,Severe injuries to R arm,Unknown
3,Australia,South Australia,Cape Jaffa Limestone Coast,Diving,?,NaN,Great White Shark 3.5m (11.5ft)
4,Australia,NSW,Little Avalon Beach,Surfing,30+,Graze injuries R leg and ankle,Unknown


In [100]:
# Función regex: extrae el primer número entero que encuentre
# \d+ = uno o más numeros consecutivos

def extraer_edad(valor):
    match = re.search(r'\d+', str(valor))
    return float(match.group()) if match else None

In [101]:
df['age'] = df['age'].apply(extraer_edad)

In [102]:
# Eliminar filas donde no se pudo extraer ningún número
df = df.dropna(subset=['age'])

# Filtrar edades imposibles
df = df[(df['age'] >= 1) & (df['age'] <= 100)]

In [103]:
print(f'Shape tras limpiar age: {df.shape}')
print(f'Rango de edades: {int(df["age"].min())} - {int(df["age"].max())} años')

Shape tras limpiar age: (4028, 7)
Rango de edades: 1 - 87 años


In [104]:
# Convertir age a entero (ya sin nulos ni valores imposibles)
df['age'] = df['age'].astype(int)

In [105]:
# Crear rangos de edad con pd.cut()
bins   = [0, 17, 25, 35, 50, 65, 100]
labels = ['<18', '18-25', '26-35', '36-50', '51-65', '65+']

In [117]:
df['age_range'] = pd.cut(df['age'], bins=bins, labels=labels, right=True)

In [107]:
print('Distribución por rango de edad:')
print(df['age_range'].value_counts().sort_index())

Distribución por rango de edad:
age_range
<18      1072
18-25    1077
26-35     804
36-50     689
51-65     312
65+        74
Name: count, dtype: int64


In [108]:
# Rellenar nulos restantes en columnas no críticas con 'Unknown'
for col in ['state', 'location', 'activity', 'species']:
    df[col] = df[col].fillna('Unknown')


In [109]:
print('Nulos restantes:')
print(df.isna().sum())

Nulos restantes:
country      9
state        0
location     0
activity     0
age          0
injury       6
species      0
age_range    0
dtype: int64


#comentario
-Solo quedan 9 nulos en country y 6 en injury. Los eliminamos porque son columnas críticas para nuestro análisis:

In [110]:
#Dado que necesitamos saber el dato de en que countries hubo ataques eliminamos los datos de los que no tenemos esa informacion
# Sin país no podemos ubicar el ataque, sin injury no sabemos si fue fatal
df = df.dropna(subset=['country', 'injury'])
print(f'Shape final: {df.shape}')

Shape final: (4013, 8)


In [111]:
#Para conseguir un ranking necesitamos una tasa de fatalidad y mortalidad con la cual poder descartar paises de forma rotunda y saber con cuales quedarnos
# Crear columna booleana 'fatal'
# True  → el texto de injury contiene la palabra 'fatal'
# False → cualquier otro caso

df['fatal'] = df['injury'].str.contains('fatal', case=False, na=False)

print('Distribución fatal vs no fatal:')
print(df['fatal'].value_counts())
print(f'\n% ataques fatales: {df["fatal"].mean()*100:.1f}%')

Distribución fatal vs no fatal:
fatal
False    3331
True      682
Name: count, dtype: int64

% ataques fatales: 17.0%


#Comentario 
-De 4017 ataques registrados, 683 fueron fatales — un 17%.

In [118]:
#VERIFICAMOS
print('=== SHAPE FINAL ===')
print(df.shape)

print('\n=== TIPOS DE DATOS ===')
print(df.dtypes)

print('\n=== NULOS RESTANTES ===')
print(df.isna().sum())

print('\n=== DUPLICADOS ===')
print(f'Duplicados: {df.duplicated().sum()}')

=== SHAPE FINAL ===
(4013, 9)

=== TIPOS DE DATOS ===
country        object
state          object
location       object
activity       object
age             int64
injury         object
species        object
age_range    category
fatal            bool
dtype: object

=== NULOS RESTANTES ===
country      0
state        0
location     0
activity     0
age          0
injury       0
species      0
age_range    0
fatal        0
dtype: int64

=== DUPLICADOS ===
Duplicados: 0


In [119]:
df.head(5)

,country,state,location,activity,age,injury,species,age_range,fatal
1,Australia,South Australia,Middleton Beach Fleurieu Peninsula Adelaide,Surfing,16,Bite wound to R ankle,Bronze Whaler,<18,False
2,Bahamas,Andros Island,Fresh Creek,Swimming,22,Severe injuries to R arm,Unknown,18-25,False
4,Australia,NSW,Little Avalon Beach,Surfing,30,Graze injuries R leg and ankle,Unknown,26-35,False
5,USA,California,Big River Beach Mendocino County,Surfing,39,Injuries to both legs and feet,Great White Shark,36-50,False
8,Australia,Queensland,Lady Elliott Island,Snorkeling,50,Lacerations elbow and abdomen,Unknown,36-50,False


#Traas la ejecuccion del documento podemos ver lo siguiente:
- falta de coherencia de formato
- caractares inusuales en las explicaciones
-En el apartado de state observamos siglas en los nombres sin poder identificar bien el lugar.

In [123]:
df['state'] = df['state'].str.replace('NSW', 'New South Wales', case=False)
df['state'] = df['state'].str.replace('Between New Ireland & New Britain', 'New Ireland / Britain', case=False)
df['state'] = df['state'].str.title()

df['injury'] = df['injury'].astype(str).str.strip()
df['injury'] = df['injury'].str.replace(r'\bR\b', 'right', case=False, regex=True)
df['injury'] = df['injury'].str.replace(r'\bL\b', 'left', case=False, regex=True)
df['injury'] = df['injury'].str.lower().str.capitalize()

print(df['injury'].unique()[:10])

['Bite wound to right ankle' 'Severe injuries to right arm'
 'Graze injuries right leg and ankle' 'Injuries to both legs and feet'
 'Lacerations elbow and abdomen' 'Deep wounds arms and legs'
 'Right thigh and lower leg stripped of flesh' 'Minor cuts and abrasions'
 'None reported damage to board' 'Serious leg injuries']


In [125]:
df.head()

,country,state,location,activity,age,injury,species,age_range,fatal
1,Australia,South Australia,Middleton Beach Fleurieu Peninsula Adelaide,Surfing,16,Bite wound to right ankle,Bronze Whaler,<18,False
2,Bahamas,Andros Island,Fresh Creek,Swimming,22,Severe injuries to right arm,Unknown,18-25,False
4,Australia,New South Wales,Little Avalon Beach,Surfing,30,Graze injuries right leg and ankle,Unknown,26-35,False
5,USA,California,Big River Beach Mendocino County,Surfing,39,Injuries to both legs and feet,Great White Shark,36-50,False
8,Australia,Queensland,Lady Elliott Island,Snorkeling,50,Lacerations elbow and abdomen,Unknown,36-50,False
